db/cache_mutation_reader.hh

```diff
@@ -228,8 +239,21 @@ public:
         , _read_time(get_read_time())
         , _gc_before(get_gc_before(*_schema, dk, _read_time))
     {
-        clogger.trace("csm {}: table={}.{}, reversed={}, snap={}", fmt::ptr(this), _schema->ks_name(), _schema->cf_name(), _read_context.is_reversed(),
-                      fmt::ptr(&*_snp));
+        if (ctx.max_purgeable_fn()) {
+            _max_purgeable_timestamp = ctx.max_purgeable_fn()(dk, is_shadowable::no);
+            _max_purgeable_timestamp_shadowable = ctx.max_purgeable_fn()(dk, is_shadowable::yes);
+        }
+
```

Comment:
Rather than having a conditional function you can make the function always exist and return min_timestamp when purging is not wanted.

Reply:
yes, instead of `max_purgeable_fn = {}`, I can default it to `never_gc` and then I can call it undonditionally

Reply:
Done in v3.